crates/cover_crypt/src/wasm_bindgen/hybrid_cc_aes.rs

```diff
@@ -340,3 +340,41 @@ pub fn webassembly_hybrid_decrypt(
     );
     Ok(Uint8Array::from(ser.finalize().as_slice()))
 }
+
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen(typescript_type = "{ encryptedHeader: Uint8Array, ciphertext: Uint8Array }")]
+    pub type EncryptedHeaderCiphertext;
+}
+
+/// Split the encrypted header from the ciphertext
+#[wasm_bindgen]
+pub fn webassembly_split_encrypted_header(
+    encrypted_bytes: Uint8Array,
+) -> Result<EncryptedHeaderCiphertext, JsValue> {
+    // Read encrypted bytes as the concatenation of an encrypted header and a DEM
+    // ciphertext.
+    let encrypted_bytes = encrypted_bytes.to_vec();
+    let mut de = Deserializer::new(&encrypted_bytes);
```

Comment:
```suggestion
    let mut de = Deserializer::new(&encrypted_bytes.to_vec());
```

Reply:
can't do that, `Deserializer` needs the `encrypted_bytes` vec to live as long as `de`, since it borrows data internally